scripts/upload-counterpart-agreements.py

```diff
@@ -33,25 +40,48 @@
 from dmscripts.helpers.env_helpers import get_api_endpoint_from_stage
 from dmscripts.bulk_upload_documents import get_bucket_name, get_all_files_of_type
 from dmscripts.upload_counterpart_agreements import upload_counterpart_file
+from dmscripts.helpers import logging_helpers
+from dmscripts.helpers.logging_helpers import logging
+
 from docopt import docopt
 from dmapiclient import DataAPIClient
 
 from dmutils.s3 import S3
+from dmutils.email.dm_notify import DMNotifyClient
+
+
+logger = logging_helpers.configure_logger({
+    'dmapiclient.base': logging.WARNING,
+    'notifications_python_client.base': logging.WARNING,
+})
 
 
 if __name__ == '__main__':
     arguments = docopt(__doc__)
 
+    if bool(arguments.get("--notify-key")) != bool(arguments.get("--notify-template-id")):
+        raise ValueError("Either specify both --notify-key and --notify-template-id or neither")
+
     stage = arguments['<stage>']
-    client = DataAPIClient(get_api_endpoint_from_stage(stage), arguments['<api_token>'])
-    framework_slug = arguments['<framework_slug>']
+    data_api_client = DataAPIClient(get_api_endpoint_from_stage(stage), arguments['<api_token>'])
+    framework = data_api_client.get_framework(arguments['<framework_slug>'])["frameworks"]
     document_directory = arguments['<documents_directory>']
     dry_run = arguments['--dry-run']
+    dm_notify_client = arguments.get("--notify-key") and DMNotifyClient(arguments["--notify-key"], logger=logger)
```

Comment:
I was in two minds over this - on one hand, what you say - on the other hand I'd like to somehow give it a chance for the API key to fail if it's going to. But I don't know if the notify client does any sort of handshaking on client creation to verify that - I imagine it doesn't. Maybe it can verify the key's formatting...?